nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/src/main/java/org/apache/nifi/elasticsearch/ElasticSearchClientServiceImpl.java

```diff
@@ -161,6 +197,11 @@ private void setupClient(final ConfigurationContext context) throws MalformedURL
                         credentialsProvider = addCredentials(null, AuthScope.ANY, username, password);
                     }
 
+                    if (apiKeyId != null && apiKey != null) {
+                        final String apiKeyAuth = Base64.getEncoder().encodeToString((apiKeyId + ":" + apiKey).getBytes(StandardCharsets.UTF_8));
```

Comment:
Recommend separating the string formatting and encoding to separate lines for better readability.